sdk/azidentity/managed_identity_client.go

```diff
@@ -154,23 +152,24 @@ func (c *managedIdentityClient) createAccessToken(res *azcore.Response) (*azcore
 	}
 }
 
-func (c *managedIdentityClient) createAuthRequest(ctx context.Context, msiType msiType, clientID string, scopes []string) (*azcore.Request, error) {
-	switch msiType {
+func (c *managedIdentityClient) createAuthRequest(ctx context.Context, clientID string, scopes []string) (*azcore.Request, error) {
+	switch c.msiType {
 	case msiTypeIMDS:
 		return c.createIMDSAuthRequest(ctx, scopes)
-	case msiTypeAppService:
+	case msiTypeAppServiceV20170901:
+		return c.createAppServiceAuthRequest(ctx, clientID, scopes)
```

Comment:
This looks to do the same thing as the other case.  If this is intentional then you can write it in one case to make it clear.

```go
case msiTypeAppServiceV20170901, msiTypeAppServiceV20190801:
```

Reply:
I have been ping ponging between having two different create request methods but decided to just leave it in the same one for now. This might change in the future though if the requests end up looking more different. I'll do what you say for now.

Reply:
Done